pkg/scheduler/internal/cache/cache.go

```diff
@@ -254,7 +261,7 @@ func (cache *schedulerCache) UpdateSnapshot(nodeSnapshot *Snapshot) error {
 		updateAllLists = true
 	}
 
-	if updateAllLists || updateNodesHavePodsWithAffinity {
+	if updateAllLists || updateNodesHavePodsWithAffinity || updateNodesHavePodsWithRequiredAntiAffinity {
```

Comment:
nit, If we pass the flags `updateNodesHavePodsWithAffinity` and `updateNodesHavePodsWithRequiredAntiAffinity`, then we only update when the specific flag is true. Since `PodsWithAffinity` is the superset of `PodsWithRequiredAntiAffinity`, so it doesn't means we need to update 'PodsWithRequiredAntiAffinity' when `updateNodesHavePodsWithAffinity` flag is true.